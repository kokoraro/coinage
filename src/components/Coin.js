import React from 'react';

const Coin = ({ coinData, currDollar, addOpen, handleDelete }) => {
  const currency = currDollar ? 'USD' : 'GBP';
  const currSymbol = currDollar ? '$' : 'Â£';
  const value = Number(coinData.price[currency]).toFixed(2);
  const price = `${currSymbol}${value}`;

  const firstDiv = addOpen ? (
    <div className="coin-delete" onClick={() => handleDelete(coinData.id)} >
      <i className="fa fa-trash"></i>
    </div>
  ) : (
    <div className="coin-id">{coinData.id}</div>
  ) 

  return (
    <li className="coin">
      {firstDiv}
      <div className="coin-code">{coinData.code}</div>
      <div className="coin-name">{coinData.name}</div>
      <div className="coin-price">{price}</div>
    </li>
  )
}

export default Coin;