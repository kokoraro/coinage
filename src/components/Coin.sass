.coin img
  width: 25px
  margin: 0 auto

.coin-delete
  color: red
  font-size: 1.3rem

.coin-name
  grid-column: 3 / 6

.coin-price
  text-align: right
  padding-right: 10px
  grid-column: 6 / 8