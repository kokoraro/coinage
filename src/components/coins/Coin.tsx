import React from 'react';
import { Coin } from '../../redux/coins/types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface CoinProps {
    coinData: Coin;
    currDollar: boolean;
    isFavouritesView: boolean
}

const formatCoinPrice = (coinData: Coin, currDollar: boolean) => {
    const currency = currDollar ? 'USD' : 'GBP';
    const currSymbol = currDollar ? '$' : 'Â£';
    const price = Number(coinData.price[currency]);
    const value = price >= 1 ? price.toFixed(2) : price.toFixed(4);

    return `${currSymbol}${value}`;
}


const CoinItem = ({
    coinData,
    currDollar,
    isFavouritesView,
}: CoinProps) => {


    return (
        <div className="coin">
            <img src={coinData.imageURL && `https://www.cryptocompare.com${coinData.imageURL}`} alt={coinData.name} />
            <div className="coin-code">{coinData.code}</div>
            <div className="coin-name">{coinData.name}</div>
            {isFavouritesView ? (
                <FontAwesomeIcon icon="star" />
            ) : (
                <div className="coin-price">{formatCoinPrice(coinData, currDollar)}</div>
            )}
        </div>
    );
};

export default CoinItem;
