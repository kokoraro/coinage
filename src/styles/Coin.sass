.coin img
  width: 25px
  margin: 0 auto

.coin-delete
  color: red
  font-size: 1.3rem
  cursor: pointer
  &:hover
    transform: scale(1.2)

.coin-name
  grid-column: 3 / 6

.coin-price
  text-align: right
  padding-right: 10px
  grid-column: 6 / 8