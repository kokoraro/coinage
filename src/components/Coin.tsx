import React from 'react';
import './Coin.sass';

interface CoinProps {
    coinData: any;
    currDollar: boolean;
    addOpen: boolean;
    handleDelete: any;
}

const Coin = ({
    coinData,
    currDollar,
    addOpen,
    handleDelete,
}: CoinProps) => {
    const currency = currDollar ? 'USD' : 'GBP';
    const currSymbol = currDollar ? '$' : 'Â£';
    const price = Number(coinData.price[currency]);
    const value = price >= 1 ? price.toFixed(2) : price.toFixed(4);

    return (
        <li className="coin">
            {addOpen ? (
                <div
                    role="button"
                    tabIndex={0}
                    className="coin-delete"
                    onClick={() => handleDelete(coinData.id)}
                >
                    <i className="fas fa-trash-alt" />
                </div>
            ) : (
                <img src={coinData.imageURL && `https://www.cryptocompare.com${coinData.imageURL}`} alt={coinData.name} />
            )}
            <div className="coin-code">{coinData.code}</div>
            <div className="coin-name">{coinData.name}</div>
            <div className="coin-price">{`${currSymbol}${value}`}</div>
        </li>
    );
};

export default Coin;
