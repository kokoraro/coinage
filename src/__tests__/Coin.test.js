import React from 'react';
import { shallow } from 'enzyme';
import Coin from '../components/Coin';

const app = shallow(<Coin />);

describe("Coin Component", () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});