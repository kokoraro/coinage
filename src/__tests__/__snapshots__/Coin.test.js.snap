// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Coin
    coinData={
      Object {
        "id": 0,
        "name": "Test",
        "price": Object {
          "GBP": 1,
          "USD": 1,
        },
      }
    }
    currDollar={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="coin-id"
        >
          0
        </div>,
        <div
          className="coin-code"
        />,
        <div
          className="coin-name"
        >
          Test
        </div>,
        <div
          className="coin-price"
        >
          $1.00
        </div>,
      ],
      "className": "coin",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "coin-id",
        },
        "ref": null,
        "rendered": 0,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "coin-code",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
          "className": "coin-name",
        },
        "ref": null,
        "rendered": "Test",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "$1.00",
          "className": "coin-price",
        },
        "ref": null,
        "rendered": "$1.00",
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="coin-id"
          >
            0
          </div>,
          <div
            className="coin-code"
          />,
          <div
            className="coin-name"
          >
            Test
          </div>,
          <div
            className="coin-price"
          >
            $1.00
          </div>,
        ],
        "className": "coin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "coin-id",
          },
          "ref": null,
          "rendered": 0,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "coin-code",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
            "className": "coin-name",
          },
          "ref": null,
          "rendered": "Test",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "$1.00",
            "className": "coin-price",
          },
          "ref": null,
          "rendered": "$1.00",
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
